packages/cli/src/utils.ts

```diff
@@ -86,25 +144,14 @@ export function exportsField(
 
   let output: Record<string, ExportsConditions> = {};
   pkg.entrypoints.forEach((entrypoint) => {
-    const esmBuild = getExportsFieldOutputPath(entrypoint, "esm");
-    const exportConditions = {
-      module: exportsFieldConfig.envConditions.size
-        ? {
-            ...(exportsFieldConfig.envConditions.has("worker") && {
-              worker: getExportsFieldOutputPath(entrypoint, "worker"),
-            }),
-            ...(exportsFieldConfig.envConditions.has("browser") && {
-              browser: getExportsFieldOutputPath(entrypoint, "browser-esm"),
-            }),
-            default: esmBuild,
-          }
-        : esmBuild,
-      default: getExportsFieldOutputPath(entrypoint, "cjs"),
-    };
-
     output[
       "." + entrypoint.name.replace(entrypoint.package.name, "")
-    ] = exportConditions;
+    ] = exportsFieldConfig.useDevProdConditions
+      ? {
+          development: entryPointExports(entrypoint, exportsFieldConfig, "dev"),
```

Comment:
what should be done about TS declarations for this? I think it's OK to not worry about this **now** but ideally it should also work with `tsconfig.json#compilerOptions.customConditions: ['development']`